tests/introspection/ApiCtorInitTest.cs

```diff
@@ -145,6 +147,11 @@ protected virtual bool Skip (PropertyInfo pi)
 		/// <param name="type">Type of the object, to be used if special logic is required.</param>
 		protected virtual void Dispose (NSObject obj, Type type)
 		{
+			//***** ApiCtorInitTest.DefaultCtorAllowed
+			//2017-01-23 15:52:09.762 introspection[4084:16658258] *** -[NSKeyedArchiver dealloc]: warning: NSKeyedArchiver deallocated without having had -finishEncoding called on it.
+			if (obj.RespondsToSelector (finishEncoding))
+				obj.PerformSelector (finishEncoding);
```

Comment:
Hmm, I'll look into it.  It was working for me locally when I made the change.  I originally just did as:
```
if (obj is NSKeyedArchiver)
    (obj as NSKeyedArchiver).FinishEncoding ();
```
I do like your version as a bit cleaner than the above option.  I opted to switch to the selector approach so it would handle any other cases that might occur in the future instead of having to special case each one.